src/visit/Analyzer.cpp

```diff
@@ -810,19 +919,41 @@ namespace cmm
         const auto modifier = EnumModifier::NO_MOD;
         const StructOrUnionContext context(currentLocality, modifier);
         const auto& structName = node.getName();
-        const auto optStructState = structTable.get(structName);
+        const auto* structDataPtr = structTable.get(structName);
 
-        if (optStructState.has_value() && *optStructState == EnumSymState::DEFINED)
+        if (structDataPtr != nullptr && structDataPtr->symState == EnumSymState::DEFINED)
         {
             std::ostringstream builder;
             builder << "struct " << structName << " is already previously defined. This violates the multiple definition rule";
             reporter.error(builder.str(), node.getLocation());
         }
 
-        else if (!optStructState.has_value())
+        else if (structDataPtr == nullptr)
         {
-            structTable.addOrUpdate(structName, EnumSymState::DEFINED);
             scope.add(structName, context);
+
+            const auto optionalBadField = node.setupFieldTable();
+
+            if (optionalBadField.has_value())
+            {
+                std::ostringstream builder;
+                builder << "Found duplicate field in struct '" << node.getName() << "' called '"
+                        << *optionalBadField << "'";
+                reporter.error(builder.str(), node.getLocation());
+            }
+
+            // Copy StructDefinitionStatementNode FieldMap to here so that we
+            // have a copy in the StructTable.
+            // TODO: Try to reduce unnecessary copying in the future.
```

Comment:
Do we want to resolve this now?

Reply:
This would require re-thinking how the StructTable is used.  More specifically and in the current state, StructTable is only used in the Analyzer and not in the Encoder.  This would mean we would need to somehow make the StructTable visible to all of the current translation unit (only).  This sounds like a bigger change than what I want to include in this PR.  Will create a new issue, link it to here, and close this comment for resolution.

Reply:
Created issue #69 for tracking this issue/enhancement.